test/plugins/templates.test.js

```diff
@@ -178,6 +208,18 @@ describe('template plugin test', () => {
             });
         });
 
+        it('returns 404 when template tag does not exist', () => {
```

Comment:
When the version does not exist, the template will not exist. So that use case should cover that.